src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/steg/vilkårsvurdering/VilkårsvurderingUtils.kt

```diff
@@ -253,3 +262,78 @@ fun finnTilOgMedDato(tilOgMed: LocalDate?, vilkårResultater: List<VilkårResult
 
     return if (skalVidereføresEnMndEkstra) tilOgMed.plusMonths(1).sisteDagIMåned() else tilOgMed.sisteDagIMåned()
 }
+
+fun validerBarnasVilkår(vilkårsvurdering: Vilkårsvurdering, barna: List<Person>) {
+    val feiler = mutableListOf<String>()
+
+    barna.map { barn ->
+        vilkårsvurdering.personResultater
+            .flatMap { it.vilkårResultater }
+            .filter { it.personResultat?.aktør == barn.aktør }
+            .forEach { vilkårResultat ->
+                val fødselsdato = barn.fødselsdato.tilDagMånedÅr()
+                val vilkårType = vilkårResultat.vilkårType
+                if (vilkårResultat.resultat == Resultat.OPPFYLT && vilkårResultat.periodeFom == null) {
+                    feiler.add("Vilkår '$vilkårType' for barn med fødselsdato $fødselsdato mangler fom dato.")
+                }
+                if (vilkårResultat.periodeFom != null && vilkårResultat.lagOgValiderPeriodeFraVilkår().fom.isBefore(barn.fødselsdato)) {
+                    feiler.add(
+                        "Vilkår '$vilkårType' for barn med fødselsdato $fødselsdato " +
+                            "har fra-og-med dato før barnets fødselsdato."
```

Comment:
Små pirk, men vi hopper i mellom fom og fra-og-med her. Hvis alt dette skal listes opp for SB eller vises i loggene, så bør det være en av dem

Reply:
Igjen samme kode som er kopiert fra ba-sak. Jeg kan endre det til fom dato, men det bør sjekkes med faglige